pkg/apis/core/v1alpha1/federatedtypeconfig_types.go

```diff
@@ -89,6 +89,35 @@ type FederatedTypeConfig struct {
 	Status FederatedTypeConfigStatus `json:"status,omitempty"`
 }
 
+// FederatedResource is a generic resource representing any federated type
```

Comment:
What is the target of this resource type? Does the intend is want to handle arbitrary resource types?

Reply:
yes, as the comment suggest FederatedResource type generically represent any arbitrary federated type.

Reply:
The sync controller is already capable of handling propogation of arbitrary types. here is the e2e test case which does that, https://github.com/kubernetes-sigs/federation-v2/blob/master/test/e2e/crud.go#L89 and there is no need to define an API for that. #60 is some additional feature to automatically create template, override & placement CRDs for a federated type.

The type `FederatedResource` which i defined in this PR represents a generic federated type which is used by the controller to Marshall/Unmarshall Json. We still need specific federated types defined like `FederatedDeployment`, `FederatedIngress` etc.. because they help validating user inputs and offer type safety.

Reply:
Thanks @shashidharatd , clear now. But still have more question, I think we do not have `placement` policy for generic resources, what is the plan for this?